lib/l1_sender/src/commands/prove.rs

```diff
@@ -128,18 +128,26 @@ impl ProofCommand {
         let stored_batch_infos: Vec<StoredBatchInfo> = self
             .batches
             .iter()
-            .map(|batch| batch.batch.batch_info.clone().into_stored())
+            .map(|batch| {
+                batch
+                    .batch
+                    .batch_info
+                    .clone()
+                    .into_stored(&batch.batch.protocol_version)
+            })
             .collect();
         // todo: awful and temporary
         let verifier_version = match self.proof.proving_execution_version() {
             // Use default verifier for fake proofs.
+            // TODO: old removed after upgrade?
```

Comment:
reasonable suggestion, we can support the same set of version that we do for prover input generation, i.e. latest and pre-latest